### PR TITLE
[PORT] Updates the Smelter UI into a TGUI format

### PR DESCRIPTION
**Pull request by Bobbanz1**
## About The Pull Request
Ports the following PR from downstream: 
* https://github.com/BeeStation/NSV13/pull/2326

Said PR literally just updates the Smelter UI into a TGUI version

## Why It's Good For The Game
No more constantly updating HTML Interface for the Smelter.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/59128051/222785911-520d9bf1-8384-4ec7-8abb-121459cceda8.mp4

Screenshot 2023-03-03 181315

</details>

## Changelog
:cl:
refactor: Refactors the Smelter User Interface into a TGUI Version
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/Smelter.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      <Table.Cell collapsing textAlign="right">
+        <Box mr={2} color="label" inline>
+          {amountAvailable} sheets
+        </Box>
+      </Table.Cell>
+      <Table.Cell collapsing>
+        <Button
+          color={material.smelting ? "good" : "bad"}
+          content={material.smelting ? "Smelting" : "Not Smelting"}
+          onClick={() => onRelease(amount)} />
```
Comment (itsmeow) [inline tgui/packages/tgui/interfaces/Smelter.js]: passing amount here does nothing

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/Smelter.js (hunk 1 above)
Comment (Bobbanz1) [reply tgui/packages/tgui/interfaces/Smelter.js]: Fixed

**Review by itsmeow — APPROVED** [review]

File: tgui/packages/tgui/interfaces/Smelter.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      </Window.Content>
+    </Window>
+  );
+};
+
+const MaterialRow = (props, context) => {
+  const { material, onRelease } = props;
+
+  const [
+    amount,
```
Comment (itsmeow) [inline tgui/packages/tgui/interfaces/Smelter.js]: this state seems unused? make sure to remove useLocalState import also

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/Smelter.js (hunk 2 above)
Comment (Bobbanz1) [reply tgui/packages/tgui/interfaces/Smelter.js]: Done

**Review by itsmeow — APPROVED** [review]

**Pull request merged**